lib/sms_autofill.dart

```diff
@@ -377,19 +367,21 @@ class TextFieldPinAutoFill extends StatefulWidget {
   final InputDecoration decoration;
   final bool obscureText;
   final TextStyle? style;
-
-  const TextFieldPinAutoFill({
-    Key? key,
-    this.focusNode,
-    this.obscureText = false,
-    this.onCodeSubmitted,
-    this.style,
-    this.onCodeChanged,
-    this.decoration = const InputDecoration(),
-    this.currentCode,
-    this.autoFocus = false,
-    this.codeLength = 6,
-  }) : super(key: key);
+  final String? smsCodeRegexPattern;
+
+  const TextFieldPinAutoFill(
+      {Key? key,
+      this.focusNode,
+      this.obscureText = false,
+      this.onCodeSubmitted,
+      this.style,
+      this.onCodeChanged,
+      this.decoration = const InputDecoration(),
+      this.currentCode,
+      this.autoFocus = false,
+      this.codeLength = 6,
```

Comment:
So now that you introduce a way to change the regex I guess `codeLength` should be taken into consideration on the default value. Because let's say I put 8 without changing the regex, I'll get only the first 6 digits only. Not sure how to handle that case. Maybe just add an assert for now saying that `smsCodeRegexPattern ` need to be modified if `codeLength` < 4 or > 6. What do you think?